### PR TITLE
feat(dashboard): style vesting overview

### LINKED ISSUE #4259: Style vesting overview
Style simple vesting overview, ignore timelocked stakes for now:
- [x] Overview pane: total vested, total locked, available rewards for collect, next payout
- [x] stake vested tokens CTA if there are no timelocked stakes 

Image

### PR DESCRIPTION
**Pull request by evavirseda**
# Description of change

Please write a summary of your changes and why you made them.

## Links to any relevant issues

fixes https://github.com/iotaledger/iota/issues/4259

## Type of change

Choose a type of change, and delete any options that are not relevant.

- Bug fix (a non-breaking change which fixes an issue)
- Enhancement (a non-breaking change which adds functionality)
- Breaking change (fix or feature that would cause existing functionality to not work as expected)
- Documentation Fix

## How the change has been tested

Describe the tests that you ran to verify your changes.

Make sure to provide instructions for the maintainer as well as any relevant configurations.

## Change checklist

Tick the boxes that are relevant to your changes, and delete any items that are not.

- [ ] I have followed the contribution guidelines for this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have checked that new and existing unit tests pass locally with my changes

### CODE REVIEW

File: apps/wallet-dashboard/lib/utils/vesting/vesting.ts [hunk 1]
```diff
@@ -28,6 +28,7 @@ import { IotaObjectData } from '@iota/iota-sdk/client';
 
 export function getLastSupplyIncreaseVestingPayout(
```
Comment (cpl121) [inline apps/wallet-dashboard/lib/utils/vesting/vesting.ts]:
```suggestion
export function getSupplyIncreaseVestingPayout(
```

File: apps/wallet-dashboard/components/staking-overview/StartStaking.tsx [hunk 2]
```diff
@@ -1,10 +1,10 @@
 // Copyright (c) 2024 IOTA Stiftung
 // SPDX-License-Identifier: Apache-2.0
 
-import { Button, ButtonSize, ButtonType, Panel } from '@iota/apps-ui-kit';
 import { StakeDialog } from '../Dialogs';
 import { Theme, useTheme } from '@iota/core';
 import { useStakeDialog } from '../Dialogs/Staking/hooks/useStakeDialog';
```
Comment (cpl121) [inline apps/wallet-dashboard/components/staking-overview/StartStaking.tsx]: we can improve the import path

File: apps/wallet-dashboard/components/Banner.tsx [hunk 3]
```diff
@@ -0,0 +1,48 @@
+// Copyright (c) 2024 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+import { Button, ButtonSize, ButtonType, Panel } from '@iota/apps-ui-kit';
+import React from 'react';
+
+interface BannerProps {
+    videoSrc: string;
+    title: string;
+    subTitle?: string;
```
Comment (cpl121) [inline apps/wallet-dashboard/components/Banner.tsx]:
```suggestion
    subtitle?: string;
```

File: apps/wallet-dashboard/app/(protected)/vesting/page.tsx [hunk 4]
```diff
@@ -43,8 +65,17 @@ function VestingDashboardPage(): JSX.Element {
     });
     const { data: timelockedStakedObjects } = useGetTimelockedStakedObjects(account?.address || '');
     const { mutateAsync: signAndExecuteTransaction } = useSignAndExecuteTransaction();
+    const { theme } = useTheme();
+
+    const [countdown, setCountdown] = useState<string | null>(null);
```
Comment (cpl121) [inline apps/wallet-dashboard/app/(protected)/vesting/page.tsx]:
```suggestion
        const [timeRemainingMs, setTimeRemainingMs] = useState<number>(0);
```

File: apps/wallet-dashboard/app/(protected)/vesting/page.tsx [hunk 5]
```diff
@@ -57,6 +88,40 @@ function VestingDashboardPage(): JSX.Element {
...
+    );
+
+    const [formattedTotalLocked, lockedSymbol] = useFormatCoin(
+        vestingSchedule.totalLocked,
+        IOTA_TYPE_ARG,
+    );
+    const lastPayout = getLastSupplyIncreaseVestingPayout([
+        ...timelockedMapped,
+        ...timelockedstakedMapped,
+    ]);
```
Comment (cpl121) [inline apps/wallet-dashboard/app/(protected)/vesting/page.tsx]:
```suggestion
    const lastPayout = getLastSupplyIncreaseVestingPayout([
        ...timelockedMapped,
        ...timelockedstakedMapped,
    ], 
    false);
```

File: apps/wallet-dashboard/app/(protected)/vesting/page.tsx [hunk 6]
```diff
@@ -57,6 +88,40 @@ function VestingDashboardPage(): JSX.Element {
...
+
+    useEffect(() => {
+        if (timeRemainingMs) {
+            const days = Math.floor(timeRemainingMs / (1000 * 60 * 60 * 24));
+            const hours = Math.floor((timeRemainingMs % (1000 * 60 * 60 * 24)) / (1000 * 60 * 60));
+            const minutes = Math.floor((timeRemainingMs % (1000 * 60 * 60)) / (1000 * 60));
+            const seconds = Math.floor((timeRemainingMs % (1000 * 60)) / 1000);
+            setCountdown(`${days} days ${hours} hours ${minutes} minutes ${seconds} seconds`);
+        }
+    }, [timeRemainingMs, countdown]);
```
Comment (cpl121) [inline apps/wallet-dashboard/app/(protected)/vesting/page.tsx]:
```suggestion
    const formatTime = (totalMilliseconds: number) => {
        const days = Math.floor(totalMilliseconds / (MILLISECONDS_PER_DAY));
        const hours = Math.floor((totalMilliseconds % (MILLISECONDS_PER_DAY)) / (MILLISECONDS_PER_HOUR));
        const minutes = Math.floor((totalMilliseconds % (MILLISECONDS_PER_HOUR)) / (MILLISECONDS_PER_MINUTE));
        const seconds = Math.floor((totalMilliseconds % (MILLISECONDS_PER_MINUTE)) / MILLISECONDS_PER_SECOND);
        return `${days} days ${hours} hours ${minutes} minutes ${seconds} seconds`;
      };

    useEffect(() => {
        if (timeRemainingMs <= 0 && lastPayout) {            
            setTimeRemainingMs(lastPayout.expirationTimestampMs && Date.now() - lastPayout.expirationTimestampMs)
        }
        const interval = setInterval(() => {
            setTimeRemainingMs((prev) => prev - MILLISECONDS_PER_SECOND);
        }, MILLISECONDS_PER_SECOND);
      
        return () => clearInterval(interval);
    }, [timeRemainingMs, lastPayout]);

    const formattedTime = formatTime(timeRemainingMs) 
```

File: apps/wallet-dashboard/app/(protected)/vesting/page.tsx [hunk 7]
```diff
@@ -142,79 +207,122 @@ function VestingDashboardPage(): JSX.Element {
...
+                            }
+                        />
+                    </Card>
+                    <Card type={CardType.Outlined}>
+                        <CardImage type={ImageType.BgSolid} shape={ImageShape.SquareRounded}>
+                            <Calendar className="h-5 w-5 text-primary-30 dark:text-primary-80" />
+                        </CardImage>
+                        <CardBody
+                            title={vestingSchedule.availableStaking.toString()}
+                            subtitle={`Next payout in ${countdown}`}
```
Comment (cpl121) [inline apps/wallet-dashboard/app/(protected)/vesting/page.tsx]:
```suggestion
                            subtitle={`Next payout in ${formattedTime}`}
```

File: apps/wallet-dashboard/app/(protected)/vesting/page.tsx [hunk 8]
```diff
@@ -142,79 +207,122 @@ function VestingDashboardPage(): JSX.Element {
...
+                        />
+                        <CardAction
+                            type={CardActionType.Button}
+                            onClick={handleCollect}
```
Comment (cpl121) [inline apps/wallet-dashboard/app/(protected)/vesting/page.tsx]: I think this button will open the schedule, it will not do a collect

**Review by cpl121 — CHANGES_REQUESTED** [review]

File: apps/wallet-dashboard/lib/utils/vesting/getMockedTimelockedStakedObjectsWithDynamicDate.ts [hunk 9]
```diff
@@ -1,7 +1,7 @@
 // Copyright (c) 2024 IOTA Stiftung
 // SPDX-License-Identifier: Apache-2.0
 
-import { MILLISECONDS_PER_DAY } from '../../constants';
+import { MILLISECONDS_PER_DAY } from '@iota/core';
```
Comment (marc2332) [inline apps/wallet-dashboard/lib/utils/vesting/getMockedTimelockedStakedObjectsWithDynamicDate.ts]: Revert this

**Review by marc2332 — CHANGES_REQUESTED** [review]

File: apps/wallet-dashboard/lib/utils/vesting/getMockedTimelockedStakedObjectsWithDynamicDate.ts (hunk 9 above)
Comment (marc2332) [reply apps/wallet-dashboard/lib/utils/vesting/getMockedTimelockedStakedObjectsWithDynamicDate.ts]: This will cause the vesting.spec.ts to import the whole @iota/core package, including JSX files which we don't really want in the tests, that's why the CI is failing

Comment (marc2332) [reply apps/wallet-dashboard/lib/utils/vesting/getMockedTimelockedStakedObjectsWithDynamicDate.ts]: If it can't be reverted, then just do something like @iota/core/dasda/constants

File: apps/wallet-dashboard/lib/constants/vesting.constants.ts [hunk 10]
```diff
@@ -1,9 +1,9 @@
 // Copyright (c) 2024 IOTA Stiftung
 // SPDX-License-Identifier: Apache-2.0
 
+import { DAYS_PER_WEEK, MILLISECONDS_PER_DAY, DAYS_PER_YEAR } from '@iota/core';
```
Comment (marc2332) [inline apps/wallet-dashboard/lib/constants/vesting.constants.ts]: Also this, use a more preceise path

**Review by marc2332 — CHANGES_REQUESTED** [review]

File: apps/wallet-dashboard/lib/utils/vesting/getMockedTimelockedStakedObjectsWithDynamicDate.ts (hunk 9 above)
Comment (evavirseda) [reply apps/wallet-dashboard/lib/utils/vesting/getMockedTimelockedStakedObjectsWithDynamicDate.ts]: fixed :pray:

File: apps/ui-kit/src/lib/components/molecules/card/CardAction.tsx [hunk 11]
```diff
@@ -2,7 +2,7 @@
 // SPDX-License-Identifier: Apache-2.0
 
 import { ArrowRight } from '@iota/ui-icons';
-import { Button, ButtonSize, ButtonType } from '@/components/atoms/button';
+import { Button, ButtonSize, ButtonType } from '../../atoms/button';
```
Comment (cpl121) [inline apps/ui-kit/src/lib/components/molecules/card/CardAction.tsx]: Why this change?

File: apps/ui-kit/src/lib/components/molecules/card/CardAction.tsx [hunk 12]
```diff
@@ -58,12 +62,13 @@ export function CardAction({
         return (
             <div className="shrink-0">
                 <Button
-                    type={ButtonType.Outlined}
```
Comment (cpl121) [inline apps/ui-kit/src/lib/components/molecules/card/CardAction.tsx:61]: why not keep this type by default?

File: apps/wallet-dashboard/lib/constants/vesting.constants.ts [hunk 13]
```diff
@@ -1,9 +1,13 @@
 // Copyright (c) 2024 IOTA Stiftung
 // SPDX-License-Identifier: Apache-2.0
 
+import {
+    DAYS_PER_WEEK,
+    MILLISECONDS_PER_DAY,
+    DAYS_PER_YEAR,
+} from '@iota/core/constants/time.constants';
```
Comment (cpl121) [inline apps/wallet-dashboard/lib/constants/vesting.constants.ts:8]:
```suggestion
} from '@iota/core';
```

File: apps/wallet-dashboard/lib/utils/vesting/getMockedTimelockedStakedObjectsWithDynamicDate.ts [hunk 14]
```diff
@@ -1,7 +1,7 @@
 // Copyright (c) 2024 IOTA Stiftung
 // SPDX-License-Identifier: Apache-2.0
 
-import { MILLISECONDS_PER_DAY } from '../../constants';
+import { MILLISECONDS_PER_DAY } from '@iota/core/constants/time.constants';
```
Comment (cpl121) [inline apps/wallet-dashboard/lib/utils/vesting/getMockedTimelockedStakedObjectsWithDynamicDate.ts:4]:
```suggestion
import { MILLISECONDS_PER_DAY } from '@iota/core';
```

File: apps/wallet-dashboard/lib/utils/vesting/vesting.ts [hunk 15]
```diff
@@ -26,8 +26,9 @@ import {
 } from '../timelock';
 import { IotaObjectData } from '@iota/iota-sdk/client';
 
-export function getLastSupplyIncreaseVestingPayout(
+export function getSupplyIncreaseVestingPayout(
```
Comment (cpl121) [inline apps/wallet-dashboard/lib/utils/vesting/vesting.ts]: I think we could improve the name of the function and the `isLastPayout` so that it is understood when the last and the first payout are obtained. wdyt?

File: apps/wallet-dashboard/tsconfig.json [hunk 16]
```diff
@@ -24,7 +24,8 @@
             "@/app/*": ["./app/*"],
             "@/components/*": ["./components/*"],
             "@/hooks/*": ["./hooks/*"],
-            "@/stores/*": ["./stores/*"]
+            "@/stores/*": ["./stores/*"],
+            "@iota/core/constants/*": ["./../core/src/constants/*"]
```
Comment (cpl121) [inline apps/wallet-dashboard/tsconfig.json:28]: why this path?

File: apps/core/src/hooks/useCountdown.ts [hunk 17]
```diff
@@ -0,0 +1,46 @@
...
+    MILLISECONDS_PER_MINUTE,
+    MILLISECONDS_PER_SECOND,
+} from '../constants';
+
+export function useCountdown(initialTimestamp: number | null): string {
+    const [timeRemainingMs, setTimeRemainingMs] = useState<number>(0);
+
+    useEffect(() => {
+        if (timeRemainingMs <= 0 && initialTimestamp) {
+            setTimeRemainingMs(initialTimestamp && Date.now() - initialTimestamp);
```
Comment (cpl121) [inline apps/core/src/hooks/useCountdown.ts]: why `Date.now() - initialTimestamp?`

File: apps/wallet-dashboard/app/(protected)/vesting/page.tsx [hunk 18]
```diff
@@ -57,6 +87,35 @@ function VestingDashboardPage(): JSX.Element {
         Number(currentEpochMs),
     );
 
+    const lastPayout = getSupplyIncreaseVestingPayout(
```
Comment (cpl121) [inline apps/wallet-dashboard/app/(protected)/vesting/page.tsx]:
```suggestion
    const nextPayout = getSupplyIncreaseVestingPayout(
```

**Review by cpl121 — CHANGES_REQUESTED** [review]

**Review by marc2332 — APPROVED** [review]

File: apps/wallet-dashboard/lib/utils/vesting/getMockedTimelockedStakedObjectsWithDynamicDate.ts (hunk 14 above)
Comment (marc2332) [reply apps/wallet-dashboard/lib/utils/vesting/getMockedTimelockedStakedObjectsWithDynamicDate.ts:4]: No

File: apps/wallet-dashboard/tsconfig.json (hunk 16 above)
Comment (marc2332) [reply apps/wallet-dashboard/tsconfig.json:28]: No

File: apps/wallet-dashboard/lib/utils/vesting/getMockedTimelockedStakedObjectsWithDynamicDate.ts (hunk 14 above)
Comment (marc2332) [reply apps/wallet-dashboard/lib/utils/vesting/getMockedTimelockedStakedObjectsWithDynamicDate.ts:4]: It was made that way to avoid loading JSX code into the vesting tests

File: apps/wallet-dashboard/tsconfig.json (hunk 16 above)
Comment (marc2332) [reply apps/wallet-dashboard/tsconfig.json:28]: It was made that way to avoid loading JSX code into the vesting tests

File: apps/wallet-dashboard/lib/constants/vesting.constants.ts (hunk 13 above)
Comment (marc2332) [reply apps/wallet-dashboard/lib/constants/vesting.constants.ts:8]: No, It was made that way to avoid loading JSX code into the vesting tests

File: apps/core/src/hooks/useCountdown.ts (hunk 17 above)
Comment (marc2332) [reply apps/core/src/hooks/useCountdown.ts]: `initialTimestamp &&` seems unnecessary

File: apps/ui-kit/src/lib/components/molecules/card/CardAction.tsx (hunk 12 above)
Comment (evavirseda) [reply apps/ui-kit/src/lib/components/molecules/card/CardAction.tsx:61]: to prevent errors :pray: https://github.com/iotaledger/iota/actions/runs/12084123164/job/33698887069#step:11:1057

File: apps/ui-kit/src/lib/components/molecules/card/CardAction.tsx [hunk 19]
```diff
@@ -58,12 +62,13 @@ export function CardAction({
...
+                    type={buttonType}
```
Comment (panteleymonchuk) [inline apps/ui-kit/src/lib/components/molecules/card/CardAction.tsx]: I had the same [error](https://github.com/iotaledger/iota/actions/runs/12084123164/job/33698887069#step:11:1057) in my PR.

And resolved it in the same way for now. But I'd set default type instead of have undefined.
```suggestion
                    type={buttonType || ButtonType.Outlined}
```
this way worked for me.

Comment (panteleymonchuk) [reply apps/ui-kit/src/lib/components/molecules/card/CardAction.tsx]: Or alternative way we can avoid using CardAction and use Button directly.

File: apps/core/src/hooks/useCountdown.ts [hunk 20]
```diff
@@ -0,0 +1,46 @@
+// Copyright (c) 2024 IOTA Stiftung
```
Comment (cpl121) [inline apps/core/src/hooks/useCountdown.ts]: modify the name of the file too to `useCountdownByTimestamp`

File: apps/wallet-dashboard/lib/utils/vesting/vesting.ts [hunk 21]
```diff
@@ -26,8 +26,9 @@ import {
 } from '../timelock';
 import { IotaObjectData } from '@iota/iota-sdk/client';
 
-export function getLastSupplyIncreaseVestingPayout(
+export function getSupplyIncreaseVestingPayoutByExpiration(
```
Comment (cpl121) [inline apps/wallet-dashboard/lib/utils/vesting/vesting.ts]:
```suggestion
export function getLatestOrEarliestSupplyIncreaseVestingPayout(
```

File: apps/wallet-dashboard/lib/utils/vesting/vesting.ts [hunk 22]
```diff
@@ -26,8 +26,9 @@ import {
...
     objects: (TimelockedObject | ExtendedDelegatedTimelockedStake)[],
+    shouldReturnLastExpiration: boolean = true,
```
Comment (cpl121) [inline apps/wallet-dashboard/lib/utils/vesting/vesting.ts]:
```suggestion
    useLastPayout: boolean = true,
```

**Review by cpl121 — CHANGES_REQUESTED** [review]

**Review by cpl121 — APPROVED** [review]

**Review by cpl121 — CHANGES_REQUESTED** [review]
image
the countdown shouldn't be negative right?

**Review by cpl121 — APPROVED** [review]

**Review by begonaalvarezd — APPROVED** [review]
